cloudinit/util.py

```diff
@@ -1802,6 +1801,24 @@ def time_rfc2822():
         ts = "??"
     return ts
 
+def boottime():
+    import ctypes
+
+    NULL_BYTES = b"\x00"
+
+    class timeval(ctypes.Structure):
+        _fields_ = [
+            ("tv_sec", ctypes.c_int64),
+            ("tv_usec", ctypes.c_int64)
+        ]
+    libc = ctypes.CDLL('/lib/libc.so.7')
+    size = ctypes.c_size_t()
+    size.value = ctypes.sizeof(timeval)
+    buf = timeval
+    if libc.sysctlbyname(b"kern.boottime" + NULL_BYTES, ctypes.byref(buf),
+                      ctypes.byref(size), None, 0) != -1:
+        return buf.value
```

Comment:
What type exactly is in the buf?  What's converting the two-field timespec into an int or float?   Let's document in the boottime() docstring the result value/format.  I suspect we'll want to convert to float to match what we get from /proc/uptime.

Reply:
uh, buf.value is wrong… this is now fixed too…